.github/workflows/build.yml

```diff
@@ -43,54 +43,112 @@ jobs:
         arch: [amd64, arm64]
     timeout-minutes: ${{ (github.event_name == 'schedule' && 90) || ((matrix.arch == 'arm64' && 45) || 30) }}
     needs: timestamp
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build slim image `--target discourse_dependencies`
+        run: |
+          cd image && ruby auto_build.rb base_slim_${{ matrix.arch }} --cache-to=type=local,dest=/tmp/build_deps_${{ matrix.arch }}
+
+      - name: Print cache
+        run: |
+          ls -lah /tmp/build_deps_${{ matrix.arch }}
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
```

Comment:
IIRC we have quite a restrictive limit on artifact storage, so I'm not sure this use of artifacts to move things between build stages is acceptable 😢 

So I think it's probably best to keep everything in  a single job

Reply:
I think I’m overthinking here, ARM and AMD are already built separately from scratch so the risk is already there.

Reply:
Yeah... agree it's not ideal, but we do already have that risk with arm/amd.

Looks like [the free storage limit](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) is 2GB across our entire organisation 😢